From:	Gertjan van Wingerde <gwingerde@gmail.com>
Subject: [PATCH 2.6.33] rt2x00: Disable powersaving for rt61pci.
Date:	Sun, 13 Dec 2009 00:39:40 +0100

We've had many reports of rt61pci failures with powersaving enabled.
Therefore, as a stop-gap measure, disable powersaving of the rt61pci
until we have found a proper solution.

Cc: stable@kernel.org
Signed-off-by: Gertjan van Wingerde <gwingerde@gmail.com>
---

John, this is an urgent fix for 2.6.33 (and stable).

---
 drivers/net/wireless/rt2x00/rt61pci.c |    4 +---
 1 files changed, 1 insertions(+), 3 deletions(-)

--- a/drivers/net/wireless/rt2x00/rt61pci.c
+++ b/drivers/net/wireless/rt2x00/rt61pci.c
@@ -2542,9 +2542,7 @@ static int rt61pci_probe_hw_mode(struct 
 	 */
 	rt2x00dev->hw->flags =
 	    IEEE80211_HW_HOST_BROADCAST_PS_BUFFERING |
-	    IEEE80211_HW_SIGNAL_DBM |
-	    IEEE80211_HW_SUPPORTS_PS |
-	    IEEE80211_HW_PS_NULLFUNC_STACK;
+	    IEEE80211_HW_SIGNAL_DBM;
 	rt2x00dev->hw->extra_tx_headroom = 0;
 
 	SET_IEEE80211_DEV(rt2x00dev->hw, rt2x00dev->dev);
